bus-mapping/src/evm/opcodes/begin_end_tx.rs

```diff
@@ -615,3 +618,57 @@ fn gen_tx_l1_fee_ops(
     )?;
     Ok(())
 }
+
+// Add two copy-events for tx access-list addresses and storage keys if EIP-2930.
+fn gen_tx_eip2930_ops(
+    state: &mut CircuitInputStateRef,
+    exec_step: &mut ExecStep,
+) -> Result<(), Error> {
+    if !state.tx.tx_type.is_eip2930_tx() {
+        return Ok(());
+    }
+
+    let tx_id = NumberOrHash::Number(state.tx_ctx.id());
+    let (address_size, storage_key_size) =
+        access_list_address_and_storage_key_sizes(&state.tx.access_list);
+
+    // Add copy event for access-list addresses.
+    let rw_counter_start = state.block_ctx.rwc;
+    state.push_copy(
```

Comment:
Do you also have a copy event if the access list is empty, i.e. `address_size == 0 && storage_key_size == 0` ?